### PR TITLE
New contributing guide

### PR DESCRIPTION
**Pull request by yknl**

### CODE REVIEW

**Comment by moxiegirl** [comment]
Do you need to explain the architecture of this repo? That might be helfpul.

Can you add an issue template to this repo along with this work?  Then, maybe just screen snap the issue template and add an Issues section before the PR section?  

https://help.github.com/articles/manually-creating-a-single-issue-template-for-your-repository/

Also, you can create the pull request template and screen cap that too;

https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+1. Describe exactly what the goal of the PR is (and link to any relevant issues)
+2. Describe how that goal was achieved through the submitted implementation.
+3. The code must be lint-free and pass the unit tests by running `$ npm run test`
+4. The code should contain [flow static type annotations](https://flow.org)
+5. Contain tests that cover any and all new functionality or code changes.
+6. Describe how the new functionality can be tested manually.
+7. Document any new features or endpoints, and describe how developers would be expected to interact with them.
+8. PR authors should agree to our contributor's agreement.
+
+Github issues marked [help-wanted](https://github.com/blockstack/blockstack.js/labels/help-wanted)
```
Comment (moxiegirl) [inline CONTRIBUTING.md]: This should go in section called *Issues*

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Maintainers
+This repo is currently maintained by Ken Liao (@yknl). 
+
+# Pull Request Reviewers
```
Comment (moxiegirl) [inline CONTRIBUTING.md]: Better organization.

`# Pull Requests`

`## Submitting`

`## Reviewers and Reviewing`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+Hank Stoever (@hstove)
```
Comment (moxiegirl) [inline CONTRIBUTING.md]: If you add all three here, the result may be they show up on every review.  You can require a number of reviewers:

https://help.github.com/articles/about-pull-request-reviews/#required-reviews

The alternative is to indicate which of these people should review.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Pull request requirements:
+
+1. Describe exactly what the goal of the PR is (and link to any relevant issues)
+2. Describe how that goal was achieved through the submitted implementation.
+3. The code must be lint-free and pass the unit tests by running `$ npm run test`
+4. The code should contain [flow static type annotations](https://flow.org)
+5. Contain tests that cover any and all new functionality or code changes.
+6. Describe how the new functionality can be tested manually.
+7. Document any new features or endpoints, and describe how developers would be expected to interact with them.
```
Comment (moxiegirl) [inline CONTRIBUTING.md]: We should also require that the build the documentation and check it. Or alternatively, build the docs as part of the automated github checks. Then, require PR reviewers to look at this check.

**Comment by yknl** [comment]
Thanks for the review, I've made edits to address the feedback. I think architecture of the repo is out of scope for the contributing guide at least for now.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+Aaron Blankstein (@kantai)
+Matthew Little (@zone117x)
+
+All pull requests require at least 2 reviewer approvals before it can be merged.
+
+# Documentation
+Documentation in this repo is automatically generated from code comments via `documentation.js`. It is the responsibility of the pull request author to verify documentation builds correctly. 
+
+# Issues
+Github issues marked [help-wanted](https://github.com/blockstack/blockstack.js/labels/help-wanted)
```
Comment (moxiegirl) [inline CONTRIBUTING.md]: Move issues before pull request --- that's the order for dev right? Find an issue, work on it, pull request.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+1. Describe exactly what the goal of the PR is (and link to any relevant issues)
+2. Describe how that goal was achieved through the submitted implementation.
+3. The code must be lint-free and pass the unit tests by running `$ npm run test`
+4. The code should contain [flow static type annotations](https://flow.org)
+5. Contain tests that cover any and all new functionality or code changes.
+6. Describe how the new functionality can be tested manually.
+7. Document any new features or endpoints, and describe how developers would be expected to interact with them. 
+8. PR authors should agree to our contributor's agreement.
+
+## Reviewers
```
Comment (moxiegirl) [inline CONTRIBUTING.md:31]: This should be subsection under REviewing pull requests right?

**Review by moxiegirl — APPROVED** [review]
Thanks @yknl

File: CONTRIBUTING.md (hunk 6 above)
Comment (yknl) [reply CONTRIBUTING.md:31]: I think it makes sense in both places, PR submitters need to know who to tag as reviewers as well.

**Pull request merged**